### ISSUE TITLE
After rebooting or updates, "d.so: object '/lib/x86_64-linux-gnu/libsnoopy.so' from /etc/ld.so.preload cannot be preloaded "

### ISSUE THREAD

**Issue by nvaiser** [a2o/snoopy#140]
## Checklist before I submit this issue report
I confirm that:
- [YES] I have tested this with the latest version of Snoopy available @ https://github.com/a2o/snoopy#i1-version-information-and-download-links (not the latest Snoopy version packaged by your favorite distribution, it might be out of date)
- [YES ] I have read Snoopy documentation @ https://github.com/a2o/snoopy/blob/master/README.md & https://github.com/a2o/snoopy/tree/master/doc
- [YES ] I have checked FAQ entries @ https://github.com/a2o/snoopy/blob/master/doc/FAQ.md
- [YES ] I have checked existing Snoopy issues for duplicates and found none @ https://github.com/a2o/snoopy/issues?q=is%3Aissue

## Issue description - Environment basics

| Key                                 | Value             | Comments                         |
| ----------------------------------- | ----------------- | ---------------------------------|
| Architecture:                       | x86_64            | |
| Linux distribution:                 | 4.18.0-parrot10-amd64            | |
| Distribution version:               |  4.18.0-parrot10-amd64 #1 SMP Debian 4.18.10-1parrot10 (2018-10-06) x86_64 GNU/Linux
            | |
| Snoopy version:                     | snoopy-2.4.6            | |
| Snoopy config file was used:        | yes /etc/snoopy.ini  (yes|no)   | |
| Snoopy threading support enabled:   | ? (yes|no)   | |

## Issue description - How was Snoopy installed
FILLIN

## Issue description - Describe the problem you are having

FILLIN

rm -f snoopy-install.sh &&
wget -O snoopy-install.sh https://github.com/a2o/snoopy/raw/install/doc/install/bin/snoopy-install.sh &&
chmod 755 snoopy-install.sh &&
./snoopy-install.sh stable

## Steps to reproduce

Unsure if it was updating the system (apt-get upgrade) or the reboot itself where this problem showed up (probably the system updates)

When running some commands as non-root user (that don't need escalation to run), such as dig:
```

 $dig @8.8.8.8 -t mx google.com
ERROR: ld.so: object '/lib/x86_64-linux-gnu/libsnoopy.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.

; <<>> DiG 9.11.4-P2-3-Debian <<>> @8.8.8.8 -t mx google.com
; (1 server found)
;; global options: +cmd
;; Got answer:
```
..etc

Running as the root user produces no error message.

## Expected result
no error message

## Actual result

```
 $dig @8.8.8.8 -t mx google.com
ERROR: ld.so: object '/lib/x86_64-linux-gnu/libsnoopy.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.

; <<>> DiG 9.11.4-P2-3-Debian <<>> @8.8.8.8 -t mx google.com
; (1 server found)
;; global options: +cmd
;; Got answer:

```

**Comment by bostjan** [comment]
@nvaiser , apparently the `/lib/x86_64-linux-gnu/libsnoopy.so` file preloader is looking for (as configured in `/etc/ld.so.preload`) isn't there, for whatever reason.

There are two options here (mostly):
- Either something changes `/etc/ld.so.preload` to point to the invalid snoopy path,
- Or the snoopy library file is removed for some reason.

I don't think this has anything to do with Snoopy itself, or am I missing something?

**Comment by nvaiser** [comment]
Thanks for the reply bostjan.

So that's the thing, the file is there so I don't know where to troubleshoot next:

#cat /etc/ld.so.preload
/lib/x86_64-linux-gnu/libsnoopy.so

 #ls -al /lib/x86_64-linux-gnu/libsnoopy.so
lrwxrwxrwx 1 root root 18 Sep 19 01:09 /lib/x86_64-linux-gnu/libsnoopy.so -> libsnoopy.so.0.0.0

#ls -al /lib/x86_64-linux-gnu/libsnoopy.so.0.0.0
-rw-r--r-- 1 root root 55984 Sep 19 01:09 /lib/x86_64-linux-gnu/libsnoopy.so.0.0.0



The also odd thing, is this error only shows up when I run a command as non-root, the same command as root produces no error.

Are the permissions/ownership set correctly on the libsnoopy.so.0.0.0 file above?

**Comment by bostjan** [comment]
Ha, interesting.
File permissions look fine.

Is this running in any sort of container, or is there some additional security measure in place (selinux, apparmor and similar)?

Can you try running your command through `strace` to see if there is anything significant returned?

    strace -f dig <YOUR-ARGS-HERE>

If you're unfamiliar with strace output, paste it somewhere (gist) and provide the link here.
In short, you should be looking at the `stat()` and `open()` calls somewhere close to the beginning of the output, and their return values.

**Comment by nvaiser** [comment]
Sorry for the delay!  Here is the strace output:

https://gist.github.com/natv/e6f4cc765262a1808b7def83845e4793


Thanks!

**Comment by bostjan** [comment]
On the first glance it looks like it's reading the file OK:
```
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libsnoopy.so", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\220W\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=55984, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8e91b09000
mmap(NULL, 58672, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f8e91afa000
mprotect(0x7f8e91aff000, 32768, PROT_NONE) = 0
mmap(0x7f8e91aff000, 20480, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x5000) = 0x7f8e91aff000
mmap(0x7f8e91b04000, 8192, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xa000) = 0x7f8e91b04000
mmap(0x7f8e91b07000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0x7f8e91b07000
close(3)                                = 0
```

Not sure what's going on. Repeating the question from my previous comment:
Is this running in any sort of container, or is there some additional security measure in place (selinux, apparmor or similar)?

**Comment by bostjan** [comment]
I've tried reproducing this in a Parrot Docker container (current `parrotsec/core:latest`), but I can't see the error.

@nvaiser, I am closing this for now (no response in 2 years), but feel free to reopen it. However, if you do so, provide a reliable method of reproduction that I can use on my side to diagnose the issue (best in a form of a Docker container, or, as a last resort, a virtual machine image).

**Comment by captainwasabi** [comment]
I am getting this error anytime I use an lxc command.  i.e. 

# lxc list

LXD runs in a snap and I get the following in the audit.log

type=AVC msg=audit(1659985925.205:660842): apparmor="DENIED" operation="open" profile="/snap/core/13425/usr/lib/snapd/snap-confine" name="/lib/x86_64-linux-gnu/libsnoopy.so.0.0.0" pid=419713 comm="snap-confine" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
